docs/README.md

```diff
@@ -998,9 +998,80 @@ _Reference_
 - https://www.freecodecamp.org/news/javascript-new-features-es2020/
 
 
+## 19. Iterators 
 
-<br/> <br/>  <br/>
+Iterators provide mechanism to iterate over collections.
 
+Following values are iterables:
+- Arrays
+- Strings
+- Maps
+- Sets
+- DOM data structures (work in progress)
+
+Plain objects are not iterable!
+
+Iterators are Objects which uses iterator protocols.
+- After defining iterator next() method is used to iterate over the collection.
+
+```
+let numbers = new Sequence(1,2,3,4,5,6);
+let iterator = numbers[Symbol.iterator]();
+
+let result = iterator.next();
+
+while(!result.done) {
+  console.log(result.value);
+  result=iterator.next()
+}
+```
+
+What is the difference between loops and iterators?
+
+- If we have to iterate over arrays we can use loops as array use indices.
+```
+let arr = [1,2,3,4]
+for(let index = 0; i< arr.length; i++){
+  console.log(arr[index])
+}
+```
+
```

Comment:
```
This kind of loop isn't always possible. For example, Lists have indices, but Sets don't, because they're unordered collections.

The other one, the foreach loop uses an Iterator behind the scenes:
```